src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
+    }
+    return payload;
```

Comment:
PullProofAI

Error handling is missing in the `validate` method.
```typescript
async validate(payload: JwtPayload) {
    try {
        const { userId } = payload;
        const client = this.redisService.getClient();
        const jwtToken = await client.get(`jwt:${userId}`);
        
        if (!jwtToken) {
            throw new UnauthorizedException();
        }
        
        return payload;
    } catch (error) {
        // Handle or log the error appropriately
        throw new UnauthorizedException();
    }
}
```

#### Issue 2: Unused Imports
1:
Unused import statements present.
```typescript
- import { Injectable, UnauthorizedException } from '@nestjs/common';
+ import { Injectable } from '@nestjs/common';
```

#### Issue 3: Redundant Constructor Parameter
11:
The `redisService` parameter in the constructor is not used.
```typescript
constructor() {
```

#### Issue 4: Incorrect Import Path
6:
Incorrect import path for `RedisService`.
```typescript
- import { RedisService } from '@liaoliaots/nestjs-redis';
+ import { RedisService } from 'nestjs-redis';
```


<!-- This is an auto-generated comment by PullProofAI -->